src/model/Game.ts

```diff
@@ -13,9 +13,25 @@ export class Game{
     }
     addPlayer(player:Player):boolean{
         if(this.numPlayers<5){
-            this.playersJoined.push(player);
-            this.numPlayers+=1;
-            return true;
+            //Player Duplication Check
+            if(!this.containsObject(player,this.playersJoined))
+            {
+                this.playersJoined.push(player);
+                this.numPlayers+=1;
+                return true;
+            }
+        }
+        return false;
+    }
+    containsObject(obj:Object, list:Array<Object>):boolean{
```

Comment:
Finding an object in an array is quite as straightforward in JS. This is only comparing the reference, not the value, which doesn't always work as expected (see [this example ](https://repl.it/repls/PleasingIdolizedProtocol). You probably need to search by id, maybe using  [`Array.prototype.find()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find)